test/test_prior.py

```diff
@@ -0,0 +1,124 @@
+
+"""
+This is for testing priors.
+"""
+
+import math
+import itertools
+import numpy as np
+import scipy.optimize as opt
+
+import pypesto
+from pypesto.objective import NegativeLogParameterPriors
+from pypesto.objective.priors import get_parameter_prior_dict
+
+
+def test_mode():
+    """
+    Tests the maximum/optimum for priors in different scales...
+    """
+
+    scales = ['lin', 'log', 'log10']
+    prior_types = ['normal', 'laplace',
+                   'logNormal']
+
+    problem_dict = {'lin': {'lb': [0], 'ub': [10], 'opt': [1]},
+                    'log': {'lb': [-3], 'ub': [3], 'opt': [0]},
+                    'log10': {'lb': [-3], 'ub': [2], 'opt': [0]}}
+
+    for prior_type, scale in itertools.product(prior_types, scales):
+
+        prior_list = [get_parameter_prior_dict(
+            0, prior_type, [1, 1], scale)]
+
+        test_prior = NegativeLogParameterPriors(prior_list)
+        test_problem = pypesto.Problem(test_prior,
+                                       lb=problem_dict[scale]['lb'],
+                                       ub=problem_dict[scale]['ub'],
+                                       dim_full=1,
+                                       x_scales=[scale])
+
+        optimizer = pypesto.ScipyOptimizer(method='Nelder-Mead')
+
+        result = pypesto.minimize(problem=test_problem,
+                                  optimizer=optimizer,
+                                  n_starts=10)
+
+        assert np.isclose(result.optimize_result.list[0]['x'],
+                          problem_dict[scale]['opt'], atol=1e-04)
+
+    # test uniform distribution:
+    for scale in scales:
+        prior_dict = get_parameter_prior_dict(
+            0, 'uniform', [1, 2], scale)
+
+        # check inside and outside of interval
+        assert abs(prior_dict['density_fun'](lin_to_scaled(.5, scale))
+                   - 0) < 1e-8
+
+        assert abs(prior_dict['density_fun'](lin_to_scaled(1.5, scale))
+                   - math.log(1)) < 1e-8
+
+        assert abs(prior_dict['density_fun'](lin_to_scaled(2.5, scale))
+                   - 0) < 1e-8
+
+
+def test_derivatives():
+    """
+    Tests the finite gradients and second order derivatives.
+    """
+
+    scales = ['lin', 'log', 'log10']
+    prior_types = ['uniform', 'normal', 'laplace', 'logNormal']
```

Comment:
please define these string constants somewhere centrally, e.g. in the constants.py, and then use them instead of string literals everywhere.